### DISCUSSION TITLE
How do I validate an image field?

### DISCUSSION THREAD

**Discussion by backpack-operations** [Laravel-Backpack/community-forum#1212] [Q&A (Help)] [answered]
So i'm using the `image` field... and people can upload an image there. But now I'm wondering... how do I validate this thing? Do I use the normal Laravel rules or does Backpack have a validation rule for it that makes it easy for me?

**Comment by jcastroa87** [comment, marked as answer]
Hello @backpack-operations 

To validate the image field in Backpack, you have a couple of options:

1. For the `upload` field type, Backpack provides a custom validation rule called `ValidUpload`. This makes validation easier, especially when dealing with create and update operations. You can use it like this:

```php
use Backpack\CRUD\app\Library\Validation\Rules\ValidUpload;

'image' => ValidUpload::field('required')
                ->file('file|mimes:jpeg,png,jpg,gif,svg|max:2048'),
```

This rule allows you to define field rules and file rules separately, making it more convenient [Docs > 6.x > Fields > FREE Field Types > upload](https://backpackforlaravel.com/docs/6.x/crud-fields).

2. However, for the `image` field type specifically, which sends a base64 string, you'll need to use a special package to validate it. Backpack doesn't currently have a built-in validation rule for the image field type. A recommended package for this is `crazybooot/base64-validation`. You can install it with:

```
composer require crazybooot/base64-validation
```

This package provides validation rules specifically for base64 encoded images [Github Issue: [Bug] The File/Image validation rule does not work on the Image field](https://github.com/Laravel-Backpack/community-forum/issues/893).

It's worth noting that the Backpack team is working on improving the uploaders system, including validation, and they're considering creating a ValidationRule for the image field type in the future [Github Issue: Image Validation](https://github.com/Laravel-Backpack/community-forum/issues/722).

Remember, always ensure you're using the appropriate validation method for the specific field type you're working with in Backpack.

Cheers.
